service/controller/clusterapi/v17/resources/machinedeploymentstatus/create.go

```diff
@@ -2,8 +2,104 @@ package machinedeploymentstatus
 
 import (
 	"context"
+
+	"github.com/giantswarm/errors/tenant"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/tenantcluster"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/controllercontext"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToMachineDeployment(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var nodes []corev1.Node
+	var ready []corev1.Node
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding nodes of tenant cluster")
+
+		l, err := cc.Client.TenantCluster.K8s.CoreV1().Nodes().List(metav1.ListOptions{})
+		if tenantcluster.IsTimeout(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "timeout fetching certificates")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+			return nil
+		} else if tenant.IsAPINotAvailable(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "tenant API not available")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		nodes = l.Items
+
+		for _, n := range nodes {
+			for _, c := range n.Status.Conditions {
+				if c.Type == corev1.NodeReady && c.Status == corev1.ConditionTrue {
+					ready = append(ready, n)
+				}
+			}
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "found nodes of tenant cluster")
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "checking if status of machine deployment needs to be updated")
+
+		replicasChanged := cr.Status.Replicas != int32(len(nodes))
+		readyReplicasChanged := cr.Status.ReadyReplicas != int32(len(ready))
+
+		if !replicasChanged && !readyReplicasChanged {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "status of machine deployment does not need to be updated")
+			return nil
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "status of machine deployment needs to be updated")
+	}
+
+	var cl v1alpha1.MachineDeployment
```

Comment:
Rather than declare `cl` as value, setting it as pointer makes sense? Because, In line 93, we use it back to the pointer. (Doesn't matter if it's best practice already)

Reply:
Nice catch buddy. Thank you.